game.py

```diff
@@ -0,0 +1,126 @@
+from pieces import *
+from board import *
+from ascii import *
+from os import system, name
+import time
+
+#need to fix knight moves and make it so it doesn't change players if an invalid move for a piece is picked
+
+def clear():
+    # for windows
+    if name == 'nt':
+      system('cls')
+    # for mac and linux(here, os.name is 'posix')
+    else:
+      system('clear')
+
+def turn(board, last_move):
+  while True:
+    clear()
+    if last_move is not None: print(last_move)
+    print("\n\n")
+    board.update_moves()
+    if board.white_to_move == True:
+      print("White's turn to move.\n\n")
+      current_player = "White"
+      opposing_player = "Black"
+    else:
+      print("Black's turn to move.\n\n")
+      current_player = "Black"
+      opposing_player = "White"
+    board.ascii.draw_board(board)
+    print("Enter ? for a list of commands or enter your move (e.g. 'move e7 to e5' or 'move g8 to f6').")
+    user_input = input()
+    if user_input == "?":
+      print("List of commands\n\n'?' for a list of commands\n'q' quits to the menu\n'draw' to request a draw\n'resign' to resign\n'save' to save the current state of the board\n'undo' to request an undo of the last move\n'board' to view the board\n'move' (e.g. 'move e7 to e5')\n")
+    elif user_input == "q":
+      return "break"
+    elif user_input == "resign":
+      while True:
+        print("Are you sure you want to resign? (y/n)")
+        resignation = input()
+        if resignation == "y":
+          print(current_player + " has resigned. " + opposing_player + " wins!")
+          time.sleep(3)
+          return "break"
+        if resignation == "n":
+          pass
+    elif user_input == "draw":
+      pass
+    elif user_input == "save":
+      pass
+    elif user_input == "undo":
+      pass
+    elif user_input == "board":
+      board.ascii.draw_board(board)
+    elif user_input.startswith("move ") and len(user_input) == 13 and user_input[5:7] in board.algebraic_squares and user_input[11:13] in board.algebraic_squares:
+      from_square = user_input[5:7]
+      to_square = user_input[11:13]
+      from_square_number = convert_to_square_number(from_square)
+      to_square_number = convert_to_square_number(to_square)
+      print(board.squares[from_square_number].piece)
+      print(board.squares[to_square_number].piece)
+      print(board.squares[from_square_number].piece_moves)
+      time.sleep(3)
+      if board.squares[from_square_number].piece.is_white == board.white_to_move:
+        board.squares[from_square_number].get_piece_moves(board)
+        print("Same color piece")
+        time.sleep(3)
+        if to_square_number in board.squares[from_square_number].piece_moves:
+          move_pieces(board, from_square_number, to_square_number)
+          print("Is white to move?")
+          print(board.white_to_move)
```

Comment:
Not in this PR, but in the future, you might consider separating out `print` statements that are going to be shown to the user in all cases as part of the value of this project, from other logging/debug messages that you want to print only to yourself while developing the code. That way you can toggle on/off debug mode so that normal people can play without this distracting info, or have it logged to a file so that you can see what happened without it being shown to normal users.

Reply:
This makes complete sense to me conceptually but am struggling a bit with how to do this practically. Can you give me a couple of additional sentences on a suggested implementation of this so I'm sure I'm going in the right direction?

Reply:
I'm not a Python expert, so I'd start with this official Python logging introduction: https://docs.python.org/3/howto/logging.html